server/src/main/java/org/opensearch/search/aggregations/MultiBucketConsumerService.java

```diff
@@ -129,11 +131,13 @@ public static class MultiBucketConsumer implements IntConsumer {
 
         // aggregations execute in a single thread so no atomic here
         private int count;
-        private int callCount = 0;
+        private LongAdder callCount;
```

Comment:
lets add a comment for `callCount`: it will be updated by multiple threads in concurrent search case so keeping it `LongAdder`

and for `count` will be called by single thread for both concurrent vs sequential case, so no atomic here

Reply:
> 

not adding the final as i am creating another constructor for testing which has the callCount variable passed.

Reply:
> lets add a comment for `callCount`: it will be updated by multiple threads in concurrent search case so keeping it `LongAdder`
> 
> and for `count` will be called by single thread for both concurrent vs sequential case, so no atomic here

done

Reply:
> lets add a comment for `callCount`: it will be updated by multiple threads in concurrent search case so keeping it `LongAdder`
> 
> and for `count` will be called by single thread for both concurrent vs sequential case, so no atomic here

done

Reply:
> constructor for testing which has the callCount variable passed.

Still `final` please, since the places it is initialized is are only constructors

Reply:
ok done